ext/har/types.go

```diff
@@ -0,0 +1,414 @@
+// Original implementation from abourget/goproxy, adapted for use as an extension.
+// HAR specification: http://www.softwareishard.com/blog/har-12-spec/
+package har
+
+import (
+    "bytes"
+    "io"
+    "log"
+    "net/http"
+    "net/url"
+    "mime"
+    "net"
+    "context"
+    "strings"
+    "time"
+)
+
+
+type Har struct {
+	Log Log `json:"log"`
+}
+
+type Log struct {
+	Version string   `json:"version"`
+	Creator Creator  `json:"creator"`
+	Browser *Browser `json:"browser,omitempty"`
+	Pages   []Page   `json:"pages,omitempty"`
+	Entries []Entry  `json:"entries"`
+	Comment string   `json:"comment,omitempty"`
+}
+
+func New() *Har {
+	har := &Har{
+		Log: Log{
+			Version: "1.2",
+			Creator: Creator{
+				Name:    "GoProxy",
+				Version: "1.0",
+			},
+			Pages:   make([]Page, 0, 10),
+			Entries: makeNewEntries(),
+		},
+	}
+	return har
+}
+
+func (har *Har) AppendEntry(entry ...Entry) {
+	har.Log.Entries = append(har.Log.Entries, entry...)
+}
+
+func (har *Har) AppendPage(page ...Page) {
+	har.Log.Pages = append(har.Log.Pages, page...)
+}
```

Comment:
Why are you exposing these functions if they are never used and the Har struct members are public?